### PR TITLE
Add sample-level README

### PR DESCRIPTION
**Pull request by annelo-msft**
Per [documentation guidance](https://aka.ms/sdkdocsoverview), sample directory should have a README describing the samples.

### CODE REVIEW

**Review by AlexGhiondea — APPROVED** [review]

File: sdk/appconfiguration/Azure.Data.AppConfiguration/samples/Sample4_ReadRevisionHistory.cs [hunk 1]
```diff
@@ -14,8 +14,8 @@ public partial class ConfigurationSamples
     {
         [Test]
         /*
-         * This sample demonstrates how to read the revision history of one or
-         * more Configuration Settings in a Configuration Store. To do this, we
+         * This sample demonstrates how to read the revision history of a
```
Comment (AlexanderSher) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/samples/Sample4_ReadRevisionHistory.cs]: Is it ok to have `a` on one line and `Configuration Settings` on another?

Comment (annelo-msft) [reply sdk/appconfiguration/Azure.Data.AppConfiguration/samples/Sample4_ReadRevisionHistory.cs]: I think it's ok, just an artifact of text wrapping.  I was trying to keep the header comment at 80 chars wide.

File: sdk/appconfiguration/Azure.Data.AppConfiguration/README.md [hunk 2]
```diff
@@ -194,14 +194,20 @@ Message: Azure.RequestFailedException : StatusCode: 404, ReasonPhrase: 'Not Foun
...
-## Next Steps
+## Next steps
 
 ### More sample code
 
 Several App Configuration client library samples are available to you in this GitHub repository.  These include: 
 - [Hello world](samples/Sample1_HelloWorld.cs): Create and delete a configuration setting.
 - [Hello world async with labels](samples/Sample2_HelloWorldExtended.cs): Asynchronously create, update and delete configuration settings with labels.
-- [Make a configuration setting readonly](samples/Sample3_SetClearReadOnly.cs): Make a configuration setting read only, and then return it to a writeable state.
+- [Make a configuration setting readonly](samples/Sample3_SetClearReadOnly.cs): Make a configuration setting read only, and then return it to a read write state.
```
Comment (heaths) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/README.md]: Nit: might say "read-write" or "read/write".

File: sdk/appconfiguration/Azure.Data.AppConfiguration/samples/README.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
+# Azure App Configuration client SDK samples
+
+This directory contains samples for the Azure App Configuration client SDK.  Each sample illustrates one or more capabilities of the library.
+
+These include:
+ - [Hello World](#hello-world): Create and delete a configuration setting.
+ - [Hello World Extended](#hello-world-extended): Asynchronously create, update and delete configuration settings with labels.
+ - [Set Read Only](#set-read-only): Make a configuration setting read only, and then return it to a read write state.
```
Comment (heaths) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/samples/README.md]: Nit: similar to above, seeing "read only" and "read write" when I more typically see these as "read-only" or "read-write", which I remember you using in your code samples I reviewed earlier today.

File: sdk/appconfiguration/Azure.Data.AppConfiguration/samples/README.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+ - [Read Revision History](#read-revision-history): Read the revision history of a configuration setting that has been changed.
+ - [Get Setting If Changed](#get-setting-if-changed): Save bandwidth by using a conditional request to retrieve a setting only if it is different from your local copy.
+ - [Update Setting If Unchanged](#update-setting-if-unchanged): Prevent lost updates by using optimistic concurrency to update a setting only if your local updates were applied to the same version as the resource in the configuration store.
+ - [Mock Client](#mock-client): Mock a client for testing using the [Moq library][moq].
+
+ ## Prerequisites
+ To run these samples, you will need:
+ - An [Azure subscription][azure_sub].
+ - A [Configuration Store][configuration_store].
+ - To set the environment variable ``"APPCONFIGURATION_CONNECTION_STRING"`` to the connection string for your configuration store.
```
Comment (heaths) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/samples/README.md]: 1. Just a single backtick for a code fragment.
2. Nit: I wouldn't put the variable in quotes after defining it as a code fragment. It's not like the variable would actually have quotes in the name, and the code fragment already sets it apart.

File: sdk/appconfiguration/Azure.Data.AppConfiguration/samples/README.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+ For more details regarding prerequisites, see the [README for this library][root_readme].
+
+## Hello World
+
+In this sample, we introduce ConfigurationClient and ConfigurationSetting, two core classes in this library.  ConfigurationClient is used to call the Azure App Configuration service -- each method call sends a request to the service's REST API.  ConfigurationSetting is the primary entity stored in a Configuration Store and represents a key-value pair you use to configure your application.  The sample walks through the basics of adding, retrieving, and deleting a configuration setting.
```
Comment (heaths) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/samples/README.md]: I'd backtick all the type and method names, like `ConfigurationClient` and `ConfigurationSetting`.

File: sdk/appconfiguration/Azure.Data.AppConfiguration/samples/README.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+[Hello World Sample](Sample1_HelloWorld.cs)
+
+## Hello World Extended
+
+In this sample, we take a step further with ConfigurationClient and show how to send requests to the service asychronously.  We also show how to create and retrieve configuration settings with labels, and provide an example scenario where labels are used to group configuration settings for beta and production application instances.
+
+[Hello World Extended Sample](Sample2_HelloWorldExtended.cs)
```
Comment (heaths) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/samples/README.md]: I love the organization here with your docs, but you might want to take a look at @pakrym's recent changes for Core. We'll probably all end up doing the same, though if you wanted to do all this now still I can't imagine a problem. Just might have to split it all apart (e.g. these are all great intros, but may need to move to separate files that use the `#region`-based snippets for which Pavel added supported).

File: sdk/appconfiguration/Azure.Data.AppConfiguration/samples/README.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+## Update Setting If Unchanged
+
+In this sample, we show how to make the SetConfigurationSetting operation conditional, so that the request it sends to the App Configuration service is a [conditional request][conditional_request_mdn].  In the Azure SDK, [conditional requests are not sent unless opted-into][conditional_request_guideline].  Making the SetConfigurationSetting operation conditional allows us to ensure that the updates we made to the configuration setting are applied only if no other clients updated the configuration setting since the last time we retrieved it.  We walk through a hypothetical scenario where we release several virtual machines from our application's resource pool, and update the ``available_vms`` configuration setting to reflect this.  In this scenario, if another client were to have modified ``available_vms`` since we last retrieved it, and we updated it unconditionally, our update would overwrite their changes and the resulting value of ``available_vms`` would be incorrect.  We show how to implement optimistic concurrency to apply the update in a way that doesn't overwrite other clients' changes.
```
Comment (heaths) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/samples/README.md]: Should only be one backtick to close and open, unless this actually is working (i.e. then nevermind).

**Review by heaths — APPROVED** [review]

**Review by heaths — APPROVED** [review]

**Pull request merged**